src/coreclr/src/jit/lsrabuild.cpp

```diff
@@ -2092,49 +2096,75 @@ void LinearScan::buildIntervals()
                 currentLoc = 1;
             }
 
-            // Any lclVars live-in to a block are resolution candidates.
-            VarSetOps::UnionD(compiler, resolutionCandidateVars, currentLiveVars);
-
-            // Determine if we need any DummyDefs.
+            // Handle special cases for live-in.
+            // If this block hasEHBoundaryIn, then we will mark the recentRefPosition of each EH Var preemptively as
+            // spillAfter, since we don't want them to remain in registers.
+            // Otherwise, determine if we need any DummyDefs.
             // We need DummyDefs for cases where "predBlock" isn't really a predecessor.
             // Note that it's possible to have uses of unitialized variables, in which case even the first
             // block may require DummyDefs, which we are not currently adding - this means that these variables
             // will always be considered to be in memory on entry (and reloaded when the use is encountered).
             // TODO-CQ: Consider how best to tune this.  Currently, if we create DummyDefs for uninitialized
             // variables (which may actually be initialized along the dynamically executed paths, but not
             // on all static paths), we wind up with excessive liveranges for some of these variables.
-            VARSET_TP newLiveIn(VarSetOps::MakeCopy(compiler, currentLiveVars));
-            if (predBlock)
-            {
-                // Compute set difference: newLiveIn = currentLiveVars - predBlock->bbLiveOut
-                VarSetOps::DiffD(compiler, newLiveIn, predBlock->bbLiveOut);
-            }
-            bool needsDummyDefs = (!VarSetOps::IsEmpty(compiler, newLiveIn) && block != compiler->fgFirstBB);
```

Comment:
Could you please explain how this PR relates to the deletion of `block != compiler->fgFirstBB` condition here?

Reply:
With this PR, we have blocks with an incoming EH boundary that may have live-in candidate vars. Previously those vars would have no live-in candidate vars, by construction, so we didn't need to worry about whether they needed dummy defs. Now that we have live-in EH vars, we don't want dummy defs for those (they should always be on stack on entry), so they are treated as having no pred block (like `fgFirstBB`), and that becomes the right condition for determining when we don't want dummy defs.